lib/utility.js

```diff
@@ -0,0 +1,7 @@
+const Utils = {
+  slugify: function(str) {
+    return str.replace(/[^A-Za-z0-9]/g, `-`).toLowerCase();
```

Comment:
Are you sure about this? Because this will make many any URL with unicode in it potentially "the same URL", and not just crazy Taiwanese name characters, but even something like a letter with an accent in someone's name, or in a library name. 

Rather than doing this, we should remove bad characters, and then replace white space and ASCII punctuation into dashes only (with consecutive dashes collapsed to a single dash). So the code that got replaces was actually *safer* than this new code =)